pkgs/development/libraries/libgudev/default.nix

```diff
@@ -18,11 +21,11 @@ stdenv.mkDerivation rec {
     sha256 = "094mgjmwgsgqrr1i0vd20ynvlkihvs3vgbmpbrhswjsrdp86j0z5";
   };
 
-  nativeBuildInputs = [ pkg-config gobject-introspection ];
-  buildInputs = [ udev glib ];
+  nativeBuildInputs = [ pkg-config gobject-introspection meson ninja ];
+  buildInputs = [ udev glib vala ];
```

Comment:
Vala is a compiler so it should likely go to `nativeBuildInputs`.

Reply:
meson was complaining about needing the a runtime dep (`vapigen`, rather than vala as a whole) so I wasn't sure. I'll fix it up.

Reply:
Yeah, vapigen is still usually used as a build tool for generating vapi files at build time.